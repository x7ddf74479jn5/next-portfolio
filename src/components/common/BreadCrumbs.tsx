import Link from "next/link";
import { useRouter } from "next/router";
import styles from "src/styles/components/common/BreadCrumbs.module.scss";
import { internalLinks } from "src/utils/paths";

const createRoutingMap = (links: typeof internalLinks): { [k: string]: string } => {
  return Object.fromEntries(
    Object.entries(links).map(([k, v]) => {
      return [v, k.toUpperCase()];
    })
  );
};

const Breadcrumbs: React.FC = () => {
  const router = useRouter();
  const pathname = router.pathname.split("/").filter((element) => {
    return element.length > 0;
  });
  const asPath = router.asPath.split("/").filter((element) => {
    return element.length > 0;
  });
  const length = pathname.length;
  const links = [];
  const routingMap = createRoutingMap(internalLinks);
  let pathnameHierarchy = "/";
  let asPathHierarchy = "/";

  for (let i = 0; i < length; i += 1) {
    const target = routingMap[pathnameHierarchy];

    links.push(
      target ? (
        <Link href={asPathHierarchy} key={asPathHierarchy}>
          <a>{target}</a>
        </Link>
      ) : (
        <Link href={asPathHierarchy} key={asPathHierarchy}>
          <a>{asPath[i]}</a>
        </Link>
      )
    );

    pathnameHierarchy += `${pathname[i]}/`;
    asPathHierarchy += `${asPath[i]}/`;
  }

  const deepest = routingMap[pathnameHierarchy]; // 最下層=現在のルーティングはクリックできないように<Typography>でリストを作成

  links.push(
    deepest ? (
      <span aria-current="page" key={asPathHierarchy}>
        {deepest}
      </span>
    ) : (
      <div key={asPathHierarchy}>{asPathHierarchy}</div>
    )
  );

  return (
    <ul className={styles.textBreadcrumbs} aria-label="Breadcrumbs">
      {links.map((link, index) => {
        return (
          <li key={index}>
            {link} {index !== links.length - 1 && <span>＞</span>}
          </li>
        );
      })}
    </ul>
  );
};

export default Breadcrumbs;
