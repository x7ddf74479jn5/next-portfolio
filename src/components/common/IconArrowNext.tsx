import Link from "next/link";

type Props = {
  label: string;
  href: string;
};

export const IconArrowNext: React.FC<Props> = ({ label, href }) => {
  return (
    <Link href={href}>
      <a className="p-btn-round p-icon__arrow-next" role="button">
        {label}
      </a>
    </Link>
  );
};
