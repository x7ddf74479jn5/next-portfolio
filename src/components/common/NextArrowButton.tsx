import clsx from "clsx";
import Link from "next/link";
import styles from "src/styles/components/common/NextArrowButton.module.scss";

type Props = {
  label?: string;
  href: string;
  onDark?: boolean;
};

export const NextArrowButton: React.FC<Props> = ({ children, onDark, label, href }) => {
  return (
    <Link href={href}>
      <a className={clsx(onDark ? styles.btnRoundOnDark : styles.btnRound, styles.iconArrowNext)} role="button">
        {children || label}
      </a>
    </Link>
  );
};
