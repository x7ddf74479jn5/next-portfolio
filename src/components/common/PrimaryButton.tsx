import styles from "src/styles/components/common/PrimaryButton.module.scss";

import type { ButtonType } from "../../types/utils/index";

type Props = ButtonType & {
  label?: string;
};

const PrimaryButton: React.FC<Props> = ({ children, disabled = false, label, ...rest }) => {
  return (
    <button className={styles.btnRound} disabled={disabled} {...rest}>
      {children || label}
    </button>
  );
};

export default PrimaryButton;
