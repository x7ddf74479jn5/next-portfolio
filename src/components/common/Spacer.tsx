import clsx from "clsx";
import styles from "src/styles/object/component/Spacer.module.scss";

const Spacer: React.VFC<Props> = ({ size }) => {
  return (
    <div
      className={clsx({
        [styles.xxxs]: size === "xxxs",
        [styles.xxs]: size === "xxs",
        [styles.xs]: size === "xs",
        [styles.sm]: size === "sm",
        [styles.md]: size === "md",
        [styles.lg]: size === "lg",
        [styles.xl]: size === "xl",
      })}
    ></div>
  );
};

export default Spacer;

type Props = {
  size: "xxxs" | "xxs" | "xs" | "sm" | "md" | "lg" | "xl";
};
