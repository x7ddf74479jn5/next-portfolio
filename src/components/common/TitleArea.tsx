import Breadcrumbs from "src/components/common/BreadCrumbs";

type Breadcrumb = {
  index: number;
  name: string;
  href: string;
};

type TitleAreaProps = {
  title: string;
  caption: string;
  // breadcrumbs: Breadcrumb[];
  // sampleId: string;
};

export const TitleArea = (props: TitleAreaProps) => {
  return (
    <section className="c-section">
      <div className="c-section__area-title animated fadeInLeft">
        <div className="p-headline__sub animated fadeInUp fast delay-1s">
          <h2>{props.title}</h2>
          <p>{props.caption}</p>
          <Breadcrumbs />
        </div>
      </div>
    </section>
  );
};
