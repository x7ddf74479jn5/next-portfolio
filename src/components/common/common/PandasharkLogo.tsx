import Image from "next/image";
import Link from "next/link";

import { images } from "../../../utils/images";

type Props = {
  width: number;
  height: number;
};

const PandasharkLogo: React.VFC<Props> = ({ width, height }) => {
  return (
    <>
      <Link href="/">
        <a target="_self" aria-label="パンダシャーク">
          <Image
            width={width}
            height={height}
            alt={images.pandasharkRectangle.alt}
            src={images.pandasharkRectangle.path}
          />
        </a>
      </Link>
    </>
  );
};

export default PandasharkLogo;
