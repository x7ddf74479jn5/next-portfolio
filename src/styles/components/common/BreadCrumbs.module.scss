@import "../../variables.scss";

.textBreadcrumbs {
  color: $main-light;
  list-style: none;
  margin-top: 2rem;
  @include mq() {
    margin-top: 3rem;
  }
  li {
    display: inline;
    font-size: $font-small;
    span {
      margin-left: 0.25rem;
      margin-right: 0.25rem;
    }
    a {
      text-decoration: none;
      transition: all 0.5s;
      &:hover {
        color: $main-dark;
        text-decoration: none;
      }
      &:visited {
        color: $main-light;
      }
    }
  }
}
