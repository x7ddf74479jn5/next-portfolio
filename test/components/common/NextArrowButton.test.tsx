import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { NextArrowButton } from "src/components/common/NextArrowButton";

describe("NextArrowButton", () => {
  const buttonLabel = "hoge";

  it("matches snapshot", () => {
    const { asFragment } = render(<NextArrowButton href="/" label={buttonLabel} />, {});
    expect(asFragment()).toMatchSnapshot();
  });

  it("has text by label", () => {
    const { container } = render(<NextArrowButton href="/" label={buttonLabel} />, {});
    expect(container).toHaveTextContent(buttonLabel);
  });

  it("has text by children", () => {
    const { container } = render(<NextArrowButton href="/">{buttonLabel}</NextArrowButton>);
    expect(container).toHaveTextContent(buttonLabel);
  });

  it("Next Link component is transformed an anchor element with appropriate href", () => {
    const { getByText } = render(<NextArrowButton href="/" label={buttonLabel} />, {});
    const targetNode = getByText(buttonLabel);
    expect(targetNode).toHaveAttribute("href", "/");
  });

  it("hover", () => {
    const { getByText } = render(<NextArrowButton href="/" label={buttonLabel} onDark />, {});
    const targetNode = getByText(buttonLabel);
    userEvent.hover(targetNode);
    expect(targetNode).toHaveClass("btnRoundOnDark");
  });
});
