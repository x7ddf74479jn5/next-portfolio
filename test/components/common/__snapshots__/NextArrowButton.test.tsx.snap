// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextArrowButton matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="btnRound iconArrowNext"
    href="/"
    role="button"
  >
    hoge
  </a>
</DocumentFragment>
`;
